File: pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/java/models/consumer/datasync/entity/PNDataSyncEntity.java
package com.pubnub.api.java.models.consumer.datasync.entity;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import lombok.experimental.Accessors;
import org.jetbrains.annotations.Nullable;

import java.util.Map;

/**
 * DataSync Entity resource returned by {@code pubnub.dataSync().getEntity()} / {@code createEntity()}.
 *
 * <p>An Entity has no default class: the caller always supplies its {@code className}.
 */
@Getter
@Setter
@NoArgsConstructor
@Accessors(chain = true)
@ToString
public class PNDataSyncEntity {
    /**
     * Entity identifier.
     */
    private String id;

    /**
     * Entity class identifier.
     */
    private String className;

    /**
     * Version of the entity class.
     */
    private int classVersion;

    /**
     * The level at which the entity class is defined (e.g. {@code SubKey} / {@code Global}).
     */
    @Nullable
    private String classLevel;

    /**
     * Date and time the entity was created.
     */
    private String createdAt;

    /**
     * Date and time the entity was last updated.
     */
    private String updatedAt;

    /**
     * The entity's content fingerprint used in conditional requests.
     */
    private String eTag;

    /**
     * Entity status.
     */
    @Nullable
    private String status;

    /**
     * Date and time when the entity expires (will be deleted automatically). Always present —
     * server-computed from the entity class TTL and never client-settable.
     */
    private String expiresAt;

    /**
     * Arbitrary user-defined JSON object.
     */
    @Nullable
    private Map<String, Object> payload;
}

File: pubnub-kotlin/pubnub-kotlin-api/src/commonMain/kotlin/com/pubnub/api/models/consumer/datasync/entity/PNDataSyncEntity.kt
package com.pubnub.api.models.consumer.datasync.entity

import com.pubnub.api.utils.SerializedName

/**
 * DataSync Entity resource returned by `pubnub.dataSync.getEntity`, `createEntity` and referenced by
 * remove operations.
 *
 * @property id Entity identifier.
 * @property className Entity class identifier.
 * @property classVersion Version of the entity class.
 * @property classLevel The level at which the entity class is defined (e.g. `SubKey` / `Global`).
 *   Modeled as a nullable [String] rather than an enum to stay source-compatible with future values.
 * @property createdAt Date and time the entity was created.
 * @property updatedAt Date and time the entity was last updated.
 * @property eTag The entity's content fingerprint used in conditional requests.
 * @property expiresAt Date and time when the entity expires (will be deleted automatically).
 *   Always present — server-computed from the entity class TTL and never client-settable.
 * @property status Entity status.
 * @property payload Arbitrary user-defined JSON object.
 */
data class PNDataSyncEntity(
    val id: String,
    @field:SerializedName("entityClass") val className: String,
    @field:SerializedName("entityClassVersion") val classVersion: Int,
    @field:SerializedName("entityClassLevel") val classLevel: String? = null,
    val createdAt: String,
    val updatedAt: String,
    val eTag: String,
    val expiresAt: String,
    val status: String? = null,
    val payload: Map<String, Any?>? = null,
)
